FindBaselibs.cmake

```diff
@@ -1,97 +1,131 @@
-set (BASEDIR /does-not-exist CACHE PATH "Path to installed baselibs _including_ OS subdirectory (Linux or Darwin).")
+set (BASEDIR "" CACHE PATH "Path to installed baselibs _including_ OS subdirectory (Linux or Darwin).")
+set (Baselibs_FOUND FALSE)
+
+if (NOT BASEDIR)
+   if (DEFINED ENV{BASELIBS})
+     set (BASEDIR $ENV{BASELIBS})
+   else()
+     message (STATUS "WARNING: BASEDIR not specified. Set BASELIBS environment variable or use cmake ... -DBASEDIR=<path>.")
+   endif()
+endif()
+
+if (BASEDIR)
+  if (IS_DIRECTORY ${BASEDIR}/lib)
+    set (Baselibs_FOUND TRUE)
+    message (STATUS "BASEDIR: ${BASEDIR}")
+  endif()
+endif()
```

Comment:
Hmm.     This seems like a step in the wrong direction.   The correct way to configure CMake is on the command line.   Env variables are a back door solution.   

If there really is no way to get UFS developers to add `-DBASEDIR=...`  on the command line,  I won't block this.    But I feel duty-bound to at least push for the correct solution.

Reply:
Yeahhhhh...I'm not a fan of this. I much prefer having `BASEDIR` set on the CMake command line. That is the CMake Way. If you want to use an environment variable for `BASEDIR` then you are probably using a script that's calling CMake. Have *that script* look for the environment variable and then set `-DBASEDIR` accordingly.

Reply:
The idea behind using `ENV{BASELIBS}` is that a module can set such an environment variable and control the build process. Modulefiles are heavily used to provide dependencies on NOAA machines. I do agree, though, that using environment variables is not encouraged in modern CMake, and I'd be happy to amend this change accordingly.